ShoppingCart-topic1/src/Demo.java

```diff
@@ -0,0 +1,92 @@
+import java.util.Scanner;
+
+public class Demo {
+
+	public static void main(String[] args) {
+		// TODO Auto-generated method stub
+
+		// Manager of the Store
+		Manager manager = new MarketManager("Amilcar", "Infante");
+		MailList.getInstance().addManager(manager);
+
+		// Products to sale
+		Item p1 = new Product("Iphone 6", 150, 300.00);
+		Item p2 = new Product("Samsung S5", 151, 250.00);
+		Item p3 = new Product("Moto X", 160, 150.00);
+		Item p4 = new Product("LG G3", 161, 225.00);
+
+		// Offers
+		Offer of1 = new Offer("Offer 1", 500, 500.00);
+		of1.addItem(p1);
+		of1.addItem(p2);
+
+		// Add the products and offers to the stock
+		Store store = new Store();
+		store.addProduct((Product) p1);

```

Comment:
To avoid the need to use a cast here you could define the method addProduct on store using Item as the type of the parameter instead of Product